Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,639 @@
+
+                # TODO(ardhi) : temporary function until GUI and core code are separated
+                def create_torrents(tor_values, _, channel_dict):
+                    """
+                    function to create torrents from channel. Adapted from
+                    ChannelManager in SearchGridManager
+                    """
+
+                    #adding new channel from the one that can't be detected from torrent values
+                    fetch_channels = set(hit[0] for hit in tor_values if hit[0] not in channel_dict)
+                    if len(fetch_channels) > 0:
+                        channels_new_dict = self.channelcast_db.getChannels(fetch_channels)
+                        channels = []
+                        for hit in channels_new_dict:
+                            channel = Channel(*hit)
+                            channels.append(channel)
+
+                        for channel in channels:
+                            channel_dict[channel.id] = channel
+
+                    # creating torrents
+                    torrents = []
+                    for hit in tor_values:
+                        if hit:
+                            chan_torrent = ChannelTorrent(*hit[1:] + [channel_dict.get(hit[0], None), None])
+                            chan_torrent.torrent_db = self.boosting_manager.torrent_db
+                            chan_torrent.channelcast_db = self.channelcast_db
+
+                            if chan_torrent.name:
+                                torrents.append(chan_torrent)
+
+                    return torrents
+
+                listtor = create_torrents(torrent_values, True,
+                                          {self.channel_id: self.channelcast_db.getChannel(self.channel_id)})
+                # listtor = self.gui_util.channelsearch_manager._createTorrents(
+                #     torrent_values, True, {self.channel_id: self.channelcast_db.getChannel(self.channel_id)})[2]
+
+                # dict {key_infohash(binary):Torrent(object-GUIDBTuple)}
+                self.unavail_torrent.update({t.infohash: t for t in listtor if t.infohash not in self.torrents})
+
+                # it's highly probable the checktor function is running at this time (if it's already running)
+                # if not running, start the checker
+                if not self.boosting_manager.is_pending_task_active(hexlify(self.source) + "_checktor"):
+                    self.boosting_manager.register_task(hexlify(self.source) + "_checktor",
+                                                        LoopingCall(self._check_tor)).start(10, now=True)
+
+    def _on_database_updated(self, subject, change_type, infohash):
+        if (subject, change_type, infohash) is None:
+            # Unused arguments
+            pass
+        self.database_updated = True
+
+    def get_source(self):
+        return self.channel.name if self.channel else None
```

Comment:
This function is used to get user-friendly source identifier. I called this function from GUI. Probably changing the name to avoid confusion